cirq-google/cirq_google/devices/known_devices.py

```diff
@@ -157,7 +157,10 @@ def create_device_proto_for_qubits(
                     # Github issue:
                     # https://github.com/quantumlib/Cirq/issues/2537
                     gate.number_of_qubits = 1
-                elif issubclass(gate_type, cirq.SingleQubitGate):
+                elif (
+                    inspect.getsource(gate_type._num_qubits_)
```

Comment:
I didn't know how check the number of qubits of a gate type now that checking it's subclass is gone except my checking how it's `num_qubits` is defined. Please let me know if there is a better way to do this.

Reply:
Hmm...`cirq.num_qubits(gate)` works for instances of the gate, but not the gate types.

Making `_num_qubits_` a `@classmethod` would resolve this, but that's a pretty significant change - not to mention the challenge of getting gates outside of Cirq to follow suit.

Reply:
And with gates like `MeasurementGate` which don't have a fixed number of qubits, a `@classmethod` becomes even more complicated, if not outright impossible.

Reply:
Given that this is only called once (during import, to generate the protos) I'm think we can get away with using `inspect` here.

Reply:
Form Cirq sync:

Have the list of gates that are in the protos set at the top of the list for Single Qubit gates instead of using `inspect`

Reply:
I've hard-coded the single qubit gates.